src/sas/qtgui/Perspectives/Fitting/FittingPerspective.py

```diff
@@ -221,8 +221,9 @@ def closeEvent(self, event):
         if self._allow_close:
             # reset the closability flag
             self.setClosable(value=False)
-            # Tell the MdiArea to close the container
-            self.parentWidget().close()
+            # Tell the MdiArea to close the container if it is visible
+            if self.parentWidget():
```

Comment:
Somewhat suspicious: the code requesting close gets here so the widget exists, but it is not bound to a parent so it doesn't get removed.

Is this a dangling pointer, where the widget was already removed, but some other function is saying to remove it again? Or is this a zombie window which might reappear later even though you told all windows to close?

The same code is in invariant and inversion, so update/ignore in all places.

Reply:
The parentWidget for a perspective will be a widget in the main window, but only if that perspective is active. All non-active perspectives are removed from the main window and held in the `self.loadedPerspectives` variable of GuiManager.